### PR TITLE
[13/N] Fix clang-tidy warnings in jit

### PR DESCRIPTION
**Pull request by cyyever**
Follows  #132209

cc @EikanWang @jgong5

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/fwd_decls.h [hunk 1]
```diff
@@ -10,7 +10,7 @@ template <typename Node>
 using NodePtr = std::shared_ptr<Node>;
 
 template <typename To, typename From>
-NodePtr<To> to(NodePtr<From> x) {
+NodePtr<To> to(const NodePtr<From>& x) {
   return std::dynamic_pointer_cast<To>(x);
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/fwd_decls.h:14]: Hmm, should it actually be a copy or move to?

Comment (cyyever) [reply torch/csrc/jit/tensorexpr/fwd_decls.h:14]: std::dynamic_pointer_cast accepts const &

File: torch/csrc/jit/tensorexpr/types.h [hunk 2]
```diff
@@ -80,7 +78,7 @@ class TORCH_API Dtype {
       std::ostream& stream,
       const Dtype& dtype);
   ScalarType scalar_type_;
-  int lanes_; // the width of the element for a vector time
+  int64_t lanes_; // the width of the element for a vector time
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/types.h:81]: Does this need to be int64_t?

Comment (cyyever) [reply torch/csrc/jit/tensorexpr/types.h:81]: Otherwise some functions for computing lanes return int64_t

**Pull request closed without merging**